Seedshop/Seedshop.cs

```diff
@@ -0,0 +1,274 @@
+			{
+				{ "price", 18 },
+				{ "available", 4 },
+				{ "panel", 6 },
+			}
+		},
+		{
+			"Rosemary",
+			new Dictionary<string, int>
+			{
+				{ "price", 25 },
+				{ "available", 2 },
+				{ "panel", 7 },
+			}
+		},
+		{
+			"Sage",
+			new Dictionary<string, int>
+			{
+				{ "price", 17 },
+				{ "available", 7 },
+				{ "panel", 8 },
+			}
+		},
+		{
+			"Chives",
+			new Dictionary<string, int>
+			{
+				{ "price", 13 },
+				{ "available", 10 },
+				{ "panel", 9 },
+			}
+		},
+		{
+			"Dill",
+			new Dictionary<string, int>
+			{
+				{ "price", 11 },
+				{ "available", 9 },
+				{ "panel", 10 },
+			}
+		},
+		{
+			"Lavender",
+			new Dictionary<string, int>
+			{
+				{ "price", 30 },
+				{ "available", 1 },
+				{ "panel", 11 },
+			}
+		},
+		{
+			"Tarragon",
+			new Dictionary<string, int>
+			{
+				{ "price", 22 },
+				{ "available", 4 },
+				{ "panel", 12 },
+			}
+		},
+		{
+			"Fennel",
+			new Dictionary<string, int>
+			{
+				{ "price", 16 },
+				{ "available", 6 },
+				{ "panel", 13 },
+			}
+		},
+		{
+			"Marjoram",
+			new Dictionary<string, int>
+			{
+				{ "price", 19 },
+				{ "available", 5 },
+				{ "panel", 14 },
+			}
+		},
+		{
+			"Lemon Balm",
+			new Dictionary<string, int>
+			{
+				{ "price", 14 },
+				{ "available", 8 },
+				{ "panel", 15 },
+			}
+		},
+		{
+			"Chervil",
+			new Dictionary<string, int>
+			{
+				{ "price", 12 },
+				{ "available", 10 },
+				{ "panel", 16 },
+			}
+		},
+	};
+
+	public override void _Ready()
+	{
+		this.Hide();
+		EventBus.Instance.OnSeedshopOpened += OpenSeedshop;
+		EventBus.Instance.OnSeedshopClosed += CloseSeedshop;
+		closeButton.Pressed += OnCloseButtonPressed;
+
+		var GridContainer = GetNode<GridContainer>("ColorRect/GridContainer");
+		if (GridContainer != null)
+		{
+			var shopSlots = GridContainer.GetChildren();
+			for (int i = 0; i < shopSlots.Count; i++)
+			{
+				var shopSlot = shopSlots[i];
+				var textureButton = shopSlot.GetNode<TextureButton>("Panel/ColorRect/TextureButton");
+				if (textureButton != null)
+				{
+					var number = i + 1;
+					textureButton.Pressed += () => SeedSold(number);
+				}
+			}
+		}
+		else
+		{
+			_logger.Error("GridContainer is null");
+		}
+
+		InitializeShopSlotLabels();
+	}
+
+	private void OpenSeedshop()
+	{
+		//work around for logging the event multiple times. On interacting, it gets called more than once
+		if (!this.Visible)
+		{
+			this.Show();
+			_logger.Debug("Seedshop opened.");
+		}
+	}
+
+	private void CloseSeedshop()
+	{
+		if (this.Visible)
+		{
+			this.Hide();
+			_logger.Debug("Seedshop closed.");
+		}
+	}
+
+	private void CustomTooltip(string name)
+	{
+		tooltip = GetNode<Label>("Seedshop/Tooltip");
+		tooltip.Text = name;
+	}
+
+	private void OnCloseButtonPressed()
+	{
+		CloseSeedshop();
+	}
+
+	private void InitializeShopSlotLabels()
+	{
+		foreach (KeyValuePair<string, Dictionary<string, int>> seed in seedData)
+		{
+			string panelName = "Panel" + seed.Value["panel"];
```

Comment:
Not priority.
Add TODO to move initialization to the slot itself.
(Equivalent of `new ShopSlot(seedData);`)